x-pack/test_serverless/api_integration/test_suites/common/data_usage/tests/metrics.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import http from 'http';
+
+import { SupertestWithRoleScope } from '@kbn/test-suites-xpack/api_integration/deployment_agnostic/services/role_scoped_supertest';
+import { UsageMetricsRequestBody } from '@kbn/data-usage-plugin/common/rest_types';
+import { FtrProviderContext } from '../../../../ftr_provider_context';
+import { setupMockServer } from '../mock_api';
+
+const API_PATH = '/internal/api/data_usage/metrics';
+export default function ({ getService }: FtrProviderContext) {
+  const svlDatastreamsHelpers = getService('svlDatastreamsHelpers');
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  let supertestAdminWithCookieCredentials: SupertestWithRoleScope;
+  const mockAutoopsApiService = setupMockServer();
+  describe('Metrics', function () {
+    let mockApiServer: http.Server;
+    // due to the plugin depending on yml config (xpack.dataUsage.enabled), we cannot test in MKI until it is on by default
+    this.tags(['skipMKI']);
+
+    before(async () => {
+      mockApiServer = mockAutoopsApiService.listen(9000);
+      supertestAdminWithCookieCredentials = await roleScopedSupertest.getSupertestWithRoleScope(
+        'admin',
+        {
+          useCookieHeader: true,
+          withInternalHeaders: true,
+        }
+      );
+    });
+
+    after(() => {
+      mockApiServer.close();
+    });
+    describe(`POST ${API_PATH}`, () => {
+      const testDataStreamName = 'test-data-stream';
+      before(async () => await svlDatastreamsHelpers.createDataStream(testDataStreamName));
+      after(async () => await svlDatastreamsHelpers.deleteDataStream(testDataStreamName));
+      it('returns 500 with non-existent data streams', async () => {
+        const requestBody: UsageMetricsRequestBody = {
+          from: 'now-24h/h',
+          to: 'now',
+          metricTypes: ['ingest_rate', 'storage_retained'],
+          dataStreams: ['invalid-data-stream'],
+        };
+        const res = await supertestAdminWithCookieCredentials
+          .post(API_PATH)
+          .set('elastic-api-version', '1')
+          .send(requestBody);
+        expect(res.statusCode).to.be(500);
+      });
+
+      it('returns 200 with valid request', async () => {
+        const requestBody: UsageMetricsRequestBody = {
+          from: 'now-24h/h',
+          to: 'now',
+          metricTypes: ['ingest_rate', 'storage_retained'],
+          dataStreams: [testDataStreamName],
+        };
+        const res = await supertestAdminWithCookieCredentials
+          .post(API_PATH)
+          .set('elastic-api-version', '1')
+          .send(requestBody);
+        expect(res.statusCode).to.be(200);
+        // TODO: decide on how to generate data
```

Comment:
I think we should have a function that takes in a date range and spits out some metrics for that range. Maybe a data point every hour?

Reply:
I'm thinking to use synthtrace